### DISCUSSION TITLE
/var/ossec/bin/verify-agent-conf allows duplicate entries but wazuh wazuh-manager.service does not

### DISCUSSION THREAD

**Discussion by biofool** [wazuh/wazuh#17704] [Ideas]
I duplicated some entries in :/var/ossec/etc/decoders/local_decoder.xml
Each entry was correctly formatted so /var/ossec/bin/verify-agent-conf reported

verify-agent-conf: Verifying [etc/shared/default/agent.conf]
verify-agent-conf: OK

But Wazuh manager reported an error.
Jun 30 05:28:26 ubuntu-kk systemd[1]: Starting Wazuh manager...
Jun 30 05:28:28 ubuntu-kk env[90711]: 2023/06/30 05:28:28 wazuh-analysisd: ERROR: (2102): Duplicated decoder with prematch: 'general_health_check1'.
Jun 30 05:28:28 ubuntu-kk env[90711]: 2023/06/30 05:28:28 wazuh-analysisd: ERROR: (2105): Error loading decoder options.
Jun 30 05:28:28 ubuntu-kk env[90711]: 2023/06/30 05:28:28 wazuh-analysisd: ERROR: (2106): Error adding decoder plugin.
Jun 30 05:28:28 ubuntu-kk env[90711]: 2023/06/30 05:28:28 wazuh-analysisd: CRITICAL: (1202): Configuration error at 'etc/decoders/local_decoder.xml'.
Jun 30 05:28:28 ubuntu-kk env[90653]: wazuh-analysisd: Configuration error. Exiting
Jun 30 05:28:28 ubuntu-kk systemd[1]: wazuh-manager.service: Control process exited, code=exited, status=1/FAILURE

**Comment by mdiego92** [comment]
Hello @biofool,

It is not possible to duplicate a decoder. Do you mind sharing the decoders you repeated so I can help you adjust that based on your requirements?

Looking forward to your feedback!

**Comment by biofool** [comment]
@mdiego92 The duplicate was a typo.  I filed the bug because verify-agent allows duplication but then wazuh fails with an error code.

So an example that is fine for verify-agent but will cause wazuh-manager to fail would be.
I can try to create a reproducible test case but I got lost tyring to find out how to file a bug report in Wazuh so this one is a little vauge:

<var name="header">myprog</var>
<var name="offset">after_parent</var>
<var name="type">syscall</var>

<decoder name="syscall">
  <prematch>^$header</prematch>
</decoder>
<decoder name="syscall">
  <prematch>^$header</prematch>
</decoder>

  <decoder name="syscall-child">
    <parent>syscall</parent>
    <prematch offset="$offset">^: $type </prematch>
    <regex offset="after_prematch">(\S+)</regex>
    <order>syscall</order>
  </decoder>

**Reply by mdiego92** [reply]
Hello @biofool,

The verify-agent-conf tool searches in /var/ossec/etc/shared the possible folders belonging to groups and verifies the agent.conf file inside them if it exists.

The error reported by the [analysisd](https://documentation.wazuh.com/current/user-manual/reference/daemons/wazuh-analysisd.html) daemon is related to a configuration error.

Could you please send me a log sample to work with it or help me understand what's your request?

Thank you very much.
Best regards!
